src/core/include/mp-units/framework/representation_concepts.h

```diff
@@ -62,105 +63,239 @@ namespace mp_units {
  */
 MP_UNITS_EXPORT enum class quantity_character : std::int8_t { scalar, complex, vector, tensor };
 
+MP_UNITS_EXPORT template<typename T>
+constexpr bool disable_scalar = false;
+
+// TODO clang-18 (and possibly other compilers) complain about the duplicated definition if `inline` is not used
+template<>
+inline constexpr bool disable_scalar<bool> = true;
```

Comment:
Yes: <https://eel.is/c++draft/temp.expl.spec#12>.

Reply:
I believe that all variable templates are implicitly `inline`. It was a recent DR, so maybe it is not implemented in Clang for now.

Reply:
Ahh, you might be right

Reply:
Wait, all global variables are `inline`: https://cplusplus.github.io/CWG/issues/2387.html. I believe that is a bug in clang.

Reply:
Although, I am not sure if I understand a note:

> [Note: An instantiated variable template that has const-qualified type can have external linkage, even if not declared extern. —end note]

Reply:
Feel free to leave it explicitly `inline`.
The only case in the C++ Standard is at the end of <https://eel.is/c++draft/fs.filesystem.syn>.